query/src/sql/exec/mod.rs

```diff
@@ -362,6 +363,12 @@ impl PipelineBuilder {
             })?;
         }
 
+        // Since transform has added or did, making group expressions as column expr is safe.
+        group_expressions = group_expressions
```

Comment:
The group expression may have been a column, so there may be duplicate conversions here.

Reply:
Yes, if a group expression is a column, it is unnecessary here.

The desitnation of adding this code is to make distinct with group-by work.
Two ways to make it better:
1. We do this only when aggregation is generated by distinct.
2. Move this when bind distinct.